openshift_runtime.py

```diff
@@ -2,9 +2,76 @@
 import time
 import shutil
 import tempfile
-import openshift_client as oc
+import subprocess
+from kubernetes import client, config
+from kubernetes.stream import stream
 from urllib.parse import urlparse
 
+# Initialize Kubernetes client
+try:
+    config.load_incluster_config()
+except:
+    config.load_kube_config()
+
+# Create API clients
+v1 = client.CoreV1Api()
+
+# Get namespace from environment variable, default to "default"
+NAMESPACE = os.getenv("COLDPRESS_NAMESPACE", "default")
+
+def copy_from_pod(pod_name, namespace, source_path, dest_path):
+    """Copy files from a pod using Kubernetes stream API instead of kubectl exec"""
+    exec_command = [
+        'tar', 'cf', '-',
+        '-C', source_path,
+        '.',
+        '--exclude', 'lost+found'
+    ]
+
+    print(f"Streaming tar data from pod {pod_name}...")
+    resp = stream(
+        v1.connect_get_namespaced_pod_exec,
+        pod_name,
+        namespace,
+        command=exec_command,
+        stderr=True,
+        stdin=False,
+        stdout=True,
+        tty=False,
+        _preload_content=False
+    )
+
+    # Start tar extraction process
+    tar_process = subprocess.Popen(
+        ['tar', 'xf', '-', '-C', dest_path],
+        stdin=subprocess.PIPE,
+        stdout=subprocess.PIPE,
+        stderr=subprocess.PIPE
+    )
+
+    try:
+        # Stream data from pod to tar process
+        while resp.is_open():
+            resp.update(timeout=1)
+            if resp.peek_stdout():
+                stdout_data = resp.read_stdout()
+                if isinstance(stdout_data, str):
+                    stdout_data = stdout_data.encode('utf-8')
+                tar_process.stdin.write(stdout_data)
+            if resp.peek_stderr():
+                stderr_data = resp.read_stderr()
+                print(f"Pod stderr: {stderr_data}")
+    finally:
+        resp.close()
+        tar_process.stdin.close()
+        tar_process.wait()
+
+    if tar_process.returncode != 0:
+        stderr_output = tar_process.stderr.read().decode()
+        raise Exception(f"tar extraction failed: {stderr_output}")
+
+    return tar_process.returncode
```

Comment:
There's no reason to return a value here: it's always going to be 0 (because if `tar_process.returncode` is not zero, we throw an exception).

Reply:
Good call, will change this.